tests/functional/spec/features/validation/sum/value_source.spec.js

```diff
@@ -0,0 +1,156 @@
+  $(SecondBreakdownAnswerPage.secondBreakdown4()).setValue(breakdown4);
+  $(SecondBreakdownAnswerPage.submit()).click();
+};
+
+describe("Feature: Sum of grouped answers equal to validation against answer value source ", () => {
+  beforeEach(() => {
+    browser.openQuestionnaire("test_validation_sum_against_value_source.json");
+  });
+
+  describe("Given I start a grouped answer validation survey and enter 12 into the total", () => {
+    it("When I continue and enter 3 in each breakdown field, Then I should be able to get to the total playback page", () => {
+      $(TotalAnswerPage.total()).setValue("12");
+      $(TotalAnswerPage.submit()).click();
+
+      answerAndSubmitBreakdownQuestion("3", "3", "3", "3");
+
+      expect(browser.getUrl()).to.contain(TotalPlaybackPage.pageName);
+    });
+  });
+
+  describe("Given I start a grouped answer validation survey and enter 12 into the total", () => {
+    it("When I continue to second breakdown and enter values equal to calculated summary total, Then I should be able to get to the summary page", () => {
+      $(TotalAnswerPage.total()).setValue("12");
+      $(TotalAnswerPage.submit()).click();
+
+      answerAndSubmitBreakdownQuestion("3", "3", "3", "3");
+
+      $(TotalPlaybackPage.submit()).click();
+
+      answerAndSubmitSecondBreakdownQuestion("2", "2", "1", "1");
+
+      expect(browser.getUrl()).to.contain(SubmitPage.pageName);
+    });
+  });
+
+  describe("Given I completed both grouped answer validation questions and I am on the summary", () => {
+    it("When I go back from the summary and change the total, Then I must reconfirm both breakdown questions with valid answers before I can get to the summary", () => {
+      $(TotalAnswerPage.total()).setValue("12");
+      $(TotalAnswerPage.submit()).click();
+
+      answerAndSubmitBreakdownQuestion("3", "3", "3", "3");
+
+      $(TotalPlaybackPage.submit()).click();
+
+      answerAndSubmitSecondBreakdownQuestion("2", "2", "1", "1");
+
+      $(SubmitPage.totalAnswerEdit()).click();
+      $(TotalAnswerPage.total()).setValue("15");
+      $(TotalAnswerPage.submit()).click();
+
+      answerAndSubmitBreakdownQuestion("6", "3", "3", "3");
+
+      $(TotalPlaybackPage.submit()).click();
+
+      answerAndSubmitSecondBreakdownQuestion("3", "3", "2", "1");
+
+      expect(browser.getUrl()).to.contain(SubmitPage.pageName);
+    });
+  });
+
+  describe("Given I completed both grouped answer validation questions and I am on the summary", () => {
+    it("When I go back from the summary and change the total, Then I must reconfirm the breakdown question based on answer value source with valid answers before I can continue", () => {
+      $(TotalAnswerPage.total()).setValue("12");
+      $(TotalAnswerPage.submit()).click();
+
+      answerAndSubmitBreakdownQuestion("3", "3", "3", "3");
+
+      $(TotalPlaybackPage.submit()).click();
+
+      answerAndSubmitSecondBreakdownQuestion("2", "2", "1", "1");
```

Comment:
You use these lines a couple of times, could be a `answer_all_questions()` method.

Reply:
Added as `asnwerBothBreakdownQuestions` since these are not all the answers and there are only two breakdown ones which we both answer here.